sdk/core/core-rest-pipeline/src/nodeHttpClient.ts

```diff
@@ -208,11 +208,12 @@ class NodeHttpClient implements HttpClient {
     };
 
     return new Promise<http.IncomingMessage>((resolve, reject) => {
-      const req = isInsecure ? http.request(options) : https.request(options);
+      const resolveCallback = (res: http.IncomingMessage) => resolve(res);
+
+      const req = isInsecure
+        ? http.request(options, resolveCallback)
+        : https.request(options, resolveCallback);
```

Comment:
No need to wrap resolve
```suggestion
      const req = isInsecure
        ? http.request(options, resolve)
        : https.request(options, resolve);
```

Reply:
Sure. :)